pkg/fleet/installer/installer.go

```diff
@@ -102,6 +103,13 @@ func (i *installerImpl) States() (map[string]repository.State, error) {
 
 // IsInstalled checks if a package is installed.
 func (i *installerImpl) IsInstalled(_ context.Context, pkg string) (bool, error) {
+	// The install script passes the package name as either <package>-<version> or <package>=<version>
+	// depending on the platform so we strip the version prefix by looking for the "real" package name
+	for _, p := range PackagesList {
+		if strings.HasPrefix(pkg, p.Name) {
+			pkg = p.Name
+		}
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 Nitpick:** ‏Missing `break`

Reply:
Also FWIW I think it can be made more robust if we use this regex https://regex101.com/r/3GcQ1w/1 as we are guaranteed that the version starts with a digit (if it is set)

Reply:
> as we are guaranteed that the version starts with a digit.

I mean, there is also `latest` as the version tag though (Which is not very recommandable to use).

I think relying on some package not being the prefix of another is equivalent if not a bit better in term of jankiness to relying on the fact that package names will never contain a number and versions will always start with one.

Reply:
For `latest` we can use the updated regex https://regex101.com/r/pmycPo/2. Given that we have a few prefixes for packages (`datadog-apm-`, `datadog-apm-library-`) already I think version is the safest route.

Another option is to error if we have more than one match.

Reply:
I mean, we could not name a new package just `datadog-apm-library` for instance, but that's not that big of an issue